packages/block-library/src/site-logo/index.php

```diff
@@ -160,3 +160,32 @@ function _delete_site_logo_on_remove_custom_logo_on_setup_theme() {
 	add_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );
 }
 add_action( 'setup_theme', '_delete_site_logo_on_remove_custom_logo_on_setup_theme', 11 );
+
+/**
+ * Removes the custom_logo theme-mod when the site_logo option gets deleted.
+ */
+function _delete_custom_logo_on_remove_site_logo() {
+	$theme = get_option( 'stylesheet' );
+
+	// Unhook update and delete actions for custom_logo to prevent a loop of hooks.
+	// Remove Gutenberg hooks.
+	remove_action( "update_option_theme_mods_$theme", '_gutenberg_delete_site_logo_on_remove_custom_logo', 10 );
+	remove_action( "delete_option_theme_mods_$theme", '_gutenberg_delete_site_logo_on_remove_theme_mods' );
+
+	// Remove Core hooks.
+	remove_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo', 10 );
+	remove_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );
+
+	// Remove the custom logo.
+	remove_theme_mod( 'custom_logo' );
+
+	// Restore update and delete actions.
+	// Restore Gutenberg hooks.
+	add_action( "update_option_theme_mods_$theme", '_gutenberg_delete_site_logo_on_remove_custom_logo', 10, 2 );
+	add_action( "delete_option_theme_mods_$theme", '_gutenberg_delete_site_logo_on_remove_theme_mods' );
+
+	// Restore Core hooks.
+	add_action( "update_option_theme_mods_$theme", '_delete_site_logo_on_remove_custom_logo', 10, 2 );
+	add_action( "delete_option_theme_mods_$theme", '_delete_site_logo_on_remove_theme_mods' );
```

Comment:
`_delete_site_logo_on_remove_custom_logo` and `_delete_site_logo_on_remove_theme_mods` both won't exist when Core's copy of `site-logo.php` [is updated](https://github.com/WordPress/wordpress-develop/pull/1804) to be the `site-logo/index.php` that we see here. This causes a fatal error.

```
Fatal error: Uncaught TypeError: call_user_func_array(): Argument #1 ($callback) must be a valid callback, function "_delete_site_logo_on_remove_custom_logo" not found or invalid function name
```

I think we should revert https://github.com/WordPress/gutenberg/pull/34820/commits/0140ba09a38367bc02215db081b3fb83ff477e25 and work around the issue with Gutenberg's build process in a different way. I'll open a PR to do this.

Reply:
https://github.com/WordPress/gutenberg/pull/34820